client-tez/src/main/java/org/apache/tez/dag/app/RssDAGAppMaster.java

```diff
@@ -269,6 +269,11 @@ public static void initAndStartRSSClient(final RssDAGAppMaster appMaster, Config
 
   @Override
   protected DAG createDAG(DAGProtos.DAGPlan dagPB) {
+    // Generally, one application will execute multiple DAGs,
+    // and there is no correlation between the DAGs.
+    // Therefore, after completing the execution of a DAG,
+    // you can unregister the relevant shuffle data.
+    tezRemoteShuffleManager.unregisterShuffle();
```

Comment:
Is it possible that we run two DAGs concurrently?

Reply:
Should we register the app again after we unregister the app?

Reply:
Even though only one current dag can run, I think we can not make sure the tez will not change in the future version.
As shuffleId is composed of dagId, source vertexId, vertexId. We should unregister all shuffle ids which are owned by dag when dag is finished, but not all shuffle id owned by app. I think this is more reasonable. Maybe the tez will support parallel dag in the future.

Reply:
@zhengchenyu hi, thank you for your reply, and I know what you mean. I have currently chosen a relatively simple way to implement it, but it can also be achieved by listening to the `DAG_FINISHED` event. If you think the latter is better, I can make changes.